[15.0][IMP+FIX] l10n_es_payment_order_confirming_aef: Cambios en archivo de confirming

victoralmau opened the pull request:
Cambios hechos:
- [x] [IMP] Añadir campo Modalidad remesa
- [x] [FIX]  Cuenta bancaria en posición correcta

1. Añadir campo Modalidad remesa

El dato relativo a la modalidad de la reserva se estaba definiendo vacío, es posible definirlo ahora según las diferentes opciones existentes según la documentación del archivo de confirming.

Links documentación:
- https://factoringasociacion.com/wp-content/uploads/2022/11/formato-estandar.pdf
- https://www.laboralkutxa.com/pdf/es/gestion-ficheros/confirming.pdf
- https://docs.bankinter.com/file_source/empresas/estaticos/stf/plataformas/empresas/gestion/ficheros/formatos_fichero/formato-aef-espanol.pdf
- https://criterium.es/sepa/generasepa/manual/confirming-estandar-AEF

2. Cuenta bancaria en posición correcta

Se debe especificar la cuenta bancaria en la posición correcta (103) por lo que se debe utilizar `ljust()` para rellenar espacios al final de la cuenta bancaria.

En la documentación se identifica: "_Los campos alfanuméricos se alinearán a la izquierda (“Nombre “) completándose con espacios por la derecha_".

@Tecnativa TT49479 TT49550

giarve committed:
[FIX] l10n_es_payment_order_confirming_aef: Santander pide IBAN entero, incluidos digitos de control

Co-authored-by: giarve

victoralmau committed:
[IMP] l10n_es_payment_order_confirming_aef: Añadir campo Modalidad remesa

El dato relativo a la modalidad de la reserva se estaba definiendo vacío,
es posible definirlo ahora según las diferentes opciones existentes según
la documentación del archivo de confirming.

TT49479

pedrobaeza commented on l10n_es_payment_order_confirming_aef/models/account_payment_mode.py:
```diff
@@ -15,6 +15,14 @@ class AccountPaymentMode(models.Model):
     )
+    aef_modality_type = fields.Selection(
+        string="Modalidad remesa",
+        default="1",
```
¿Esto es algo específico de un banco? ¿No puede ser como hasta ahora vacío también una posibilidad? ¿Puede haber más valores según el estándar?

victoralmau replied:
Ya está modificado para que por defecto la opción esté vacía y se han indicado todos los posibles valores.

En la descripción del PR se han añadido los links a la documentación de diferentes entidades bancarias (y de factoringasociacion.com) y aunque es un campo no obligatorio no se indica que se permita vacío; se añade lo relativo a este dato a la configuración y se indica que se consulte con la entidad bancaria puesto que algunos solo permiten un valor.

pedrobaeza commented on l10n_es_payment_order_confirming_aef/models/account_payment_mode.py:
```diff
@@ -15,6 +16,17 @@ class AccountPaymentMode(models.Model):
     )
+    aef_confirming_modality = fields.Selection(
+        string="Modalidad remesa",
+        default="",
```
Mejor no poner nunca un valor de cadena vacía. En v16+ da problemas con OWL. Deja `False` como valor por defecto, y al exportar haces `aef_confirming_modality or ""`.

victoralmau committed:
[FIX] l10n_es_payment_order_confirming_aef: Cuenta bancaria en posición correcta

Se debe especificar la cuenta bancaria en la posición correcta (103) por lo que
se debe utilizar ljust() para rellenar espacios al final de la cuenta bancaria.

En la documentación se identifica: "Los campos alfanuméricos se alinearán a la
izquierda (“Nombre“) completándose con espacios por la derecha".

TT49550

pedrobaeza reviewed (approved):
/ocabot merge minor

pedrobaeza commented:
/ocabot merge minor

pedrobaeza commented:
/ocabot merge minor

pedrobaeza commented:
/ocabot merge minor
